include/ares.h

```diff
@@ -224,6 +224,9 @@ typedef enum {
 #define ARES_OPT_MAXTIMEOUTMS    (1 << 20)
 #define ARES_OPT_QUERY_CACHE     (1 << 21)
 #define ARES_OPT_EVENT_THREAD    (1 << 22)
+#define ARES_OPT_SERVER_FAIL     (1 << 23)
+#define ARES_OPT_SERVER_RECOVER  (1 << 24)
+#define ARES_OPT_PRIORITY_SERVER (1 << 25)
 
```

Comment:
Not sure I'm comfortable adding 3 new options since we're running out. We only have 31 bits to work with.   Can we instead reduce this to 1 option that requires you to set all 3 settings?  Basically make a struct a member of ares_options.  Maybe just `ARES_OPT_SERVER_FAILOVER` to control failover settings.

Reply:
Thank you, I have reworked the server healthy concept (will comment in detail on another thread). I still needed 3 options for server failover behavior, but combined then into a single opt `ARES_OPT_SERVER_FAILOVER` with struct `ares_server_failover_options`.